python/ray/workflow/workflow_access.py

```diff
@@ -205,136 +64,180 @@ def gen_step_id(self, workflow_id: str, step_name: str) -> str:
-                    "please resume it"
-                )
-
-        if name is None:
-            step_id = wf_store.get_entrypoint_step_id()
-        else:
-            step_id = name
-            output = self.get_cached_step_output(workflow_id, step_id)
-            if output is not None:
-                return WorkflowStaticRef.from_output(step_id, output)
-
-        @ray.remote
-        def load(wf_store, workflow_id, step_id):
-            result = wf_store.inspect_step(step_id)
-            if result.output_object_valid:
-                # we already have the output
-                return wf_store.load_step_output(step_id)
-            if isinstance(result.output_step_id, str):
-                actor = get_management_actor()
-                return WorkflowStaticRef.from_output(
-                    result.output_step_id,
-                    actor.get_output.remote(workflow_id, result.output_step_id),
-                )
-            raise ValueError(
-                f"Cannot load output from step id {step_id} "
-                f"in workflow {workflow_id}"
-            )
-
-        return WorkflowStaticRef.from_output(
-            step_id,
-            load.remote(wf_store, workflow_id, step_id),
+        wf_store.update_workflow_status(WorkflowStatus.RUNNING)
+        logger.info(f"Workflow job [id={workflow_id}] started.")
+        try:
+            await workflow_executor.execute_workflow(job_id, context, state, wf_store)
+            # output_ref = state.get_input(state.output_task_id)
+            wf_store.update_workflow_status(WorkflowStatus.SUCCESSFUL)
+            logger.info(f"Workflow '{workflow_id}' completes successfully.")
+
+            # set errors for pending workflow outputs
+            for task_id, fut in self._workflow_output_callbacks[workflow_id].items():
+                if not fut.done():
+                    if state.is_untracked_task(task_id):
+                        # A uncommon edge case during recovery:
+                        # The output of a previous successful task is in storage
+                        # but not tracked in the state, but "workflow.get_output"
+                        # is called on the task.
+                        # TODO(suquark): Maybe we should check the storage first
+                        #   for every "workflow.get_output"? This may affect the
+                        #   performance.
+                        tid = wf_store.inspect_output(task_id)
+                        if tid is not None:
+                            fut.set_result(WorkflowRef(tid))
+                            continue
+
+                    err = RuntimeError(
+                        f"The workflow haven't yet produced output of task '{task_id}' "
+                        f"after workflow execution completes."
+                    )
+                    fut.set_exception(err)
+        except WorkflowCancellationError as e:
+            wf_store.update_workflow_status(WorkflowStatus.CANCELED)
```

Comment:
I have some concern about the storage here. I'm OK with another PR to fix this later. But given this is an eventloop, we probably don't want io operations as a blocking one.